homeassistant/components/amcrest/camera.py

```diff
@@ -92,12 +106,129 @@ def __init__(self, hass, amcrest):
         finally:
             await stream.close()
 
+    # Entity property overrides
+
     @property
     def name(self):
         """Return the name of this camera."""
         return self._name
 
+    @property
+    def device_state_attributes(self):
+        """Return the Amcrest-spectific camera state attributes."""
+        return self._static_attrs
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_ON_OFF
+
+    # Camera property overrides
+
+    @property
+    def is_recording(self):
+        """Return true if the device is recording."""
+        return self._is_recording
+
+    @property
+    def brand(self):
+        """Return the camera brand."""
+        return 'Amcrest'
+
+    @property
+    def model(self):
+        """Return the camera model."""
+        return self._model
+
     @property
     def stream_source(self):
         """Return the source of the stream."""
         return self._camera.rtsp_url(typeno=self._resolution)
+
+    @property
+    def is_on(self):
+        """Return true if on."""
+        return self.video_enabled
+
+    # Additional Amcrest Camera properties
+
+    @property
+    def video_enabled(self):
+        """Return the camera video streaming status."""
+        return self.is_streaming
+
+    @video_enabled.setter
+    def video_enabled(self, enable):
+        """Enable or disable camera video stream."""
+        from amcrest import AmcrestError
+
+        try:
+            self._camera.video_enabled = enable
+        except AmcrestError as error:
+            _LOGGER.error(
+                'Could not %s %s camera video stream due to error: %s',
+                'enable' if enable else 'disable', self.name, error)
+        else:
+            self.is_streaming = enable
+            self.schedule_update_ha_state()
+
+    # Other Entity method overrides
+
+    def update(self):
+        """Update entity status."""
+        from amcrest import AmcrestError
+
+        _LOGGER.debug('Pulling data from %s camera', self.name)
+        try:
+            if self._model is None:
+                self._model = _extract_attr(self._get_cam_attr('device_type'))
+            if not self._static_attrs:
+                self._update_static_attrs()
+            self.is_streaming = self._camera.video_enabled
+            self._is_recording = self._camera.record_mode == 'Manual'
```

Comment:
Move this to the property implementation and eliminate the backing field.

Reply:
Now you have me confused. First, I thought you told me to remove the video_enabled property. Also I thought it was a rule that data from the device should only be retrieved in the update method. Lastly, this component mostly operates in an optimistic mode; the inherited should_poll method returns False and the update method is only called once at startup to retrieve the current values from the device. Thereafter the attributes are only updated in response to services. If I do as you suggest that would completely defeat the optimistic behavior. Which reminds me - should I add assumed_state and have it return True? (I didn't think so since the Camera class didn't, even though it's should_poll method returns False.)

Reply:
> Also I thought it was a rule that data from the device should only be retrieved in the update method

That is not a rule I'm aware of.  This pattern is used extensively across platforms.

> Lastly, this component mostly operates in an optimistic mode; the inherited should_poll method returns False and the update method is only called once at startup to retrieve the current values from the device. Thereafter the attributes are only updated in response to services. If I do as you suggest that would completely defeat the optimistic behavior

Correct that `update` will only be called when added to HASS, but the state of the entity will be computed anytime `schedule_update_ha_state` is called, which will pull in attributes and properties of the entity.  Unless `record_mode` can only be retrieved once at startup, moving it to the property directly will not change the behavior.

Reply:
Sorry, when I said "data from the device" I meant actually fetching data from the physical device via the network, not simply retrieving information from the device object that represents the physical device.

The `self._camera` properties `video_enabled` and `record_mode` both fetch data from the device via the network. That is how the amcrest package was written. It actually requires a bit of an overhaul for this and many other reasons, but that's outside the scope of this PR.

So, given that, if I understand correctly, the call to these property methods should stay in update and I should continue to use the `is_streaming` and `_is_recording` attributes (which I assume is what you meant by "backing field.")